nc-news/src/App.jsx

```diff
@@ -1,40 +1,49 @@
 import { useState, useEffect } from "react";
 import { Routes, Route } from "react-router-dom";
-import Header from "./components/Header";
-import NavBar from "./components/NavBar";
-import SideBar from "./components/SideBar";
-import AllArticles from "./components/AllArticles";
-import Profile from "./components/Profile";
-import About from "./components/About";
+import {
+  NavBar,
+  SideBar,
+  AllArticles,
+  Profile,
+  About,
+  SingleArticle,
+} from "./index";
+
 import { fetchAllArticles } from "../utils/utils";
 
 function App() {
-  const [tripleBarStatus, setTripleBarStatus] = useState("header--inactive");
   const [sideBarStatus, setSideBarStatus] = useState("sidebar--inactive");
   const [allArticles, setAllArticles] = useState([]);
+  const [currentArticle, setCurrentArticle] = useState([]);
 
   useEffect(() => {
     fetchAllArticles().then((res) => {
       setAllArticles(res);
     });
   }, []);
 
-  {
-    /* 
-  </header> */
-  }
-
   return (
     <main className="container">
-      <header className={tripleBarStatus}>
-        <Header tripleBarStatus={tripleBarStatus} />
-        <NavBar setTripleBarStatus={setTripleBarStatus} />
+      <header className="header">
+        <NavBar />
       </header>
       <nav className={sideBarStatus}>
         <SideBar setSideBarStatus={setSideBarStatus} />
       </nav>
       <Routes>
-        <Route path="/" element={<AllArticles allArticles={allArticles} />} />
+        <Route
+          path="/"
+          element={
+            <AllArticles
+              allArticles={allArticles}
+              setCurrentArticle={setCurrentArticle}
+            />
+          }
+        />
+        <Route
+          path="/SingleArticle"
+          element={<SingleArticle currentArticle={currentArticle} />}
```

Comment:
I don't think the page for a single article and all articles should be sharing a state here. Mainly because these two pages have two completely different bits of data that they care about

I also think the route for a single article needs to be more specific than '/SingleArticle', it needs to tell a user which article they're on and it needs to be shareable so that a user could in essence share a link to a specific article. I would recommend '/articles/2345' or something similar with the article ID in the URL as an alternative to show the user which article they're on right now. This will have the knock-on impact of making your data fetching easier as it will give you a parameter to use in SingleArticle which will let you get a Article by its ID from the API